routes/api/v1.js

```diff
@@ -1079,6 +1079,88 @@ api.declare({
 });
 
 
+/** Report task canceled */
+api.declare({
+  method:     'post',
+  route:      '/task/:taskId/runs/:runId/canceled',
+  name:       'reportCanceled',
+  scopes: [
+    [
+      'queue:resolve-task',
+      'assume:worker-id:<workerGroup>/<workerId>'
+    ]
+  ],
+  deferAuth:  true,
+  input:      undefined,  // No input at this point
+  output:     SCHEMA_PREFIX_CONST + 'task-status-response.json#',
+  title:      "Report Task Canceled",
+  description: [
+    "Report a task as canceled, resolving the run as `exception`. Use this to cancel",
+    "a run that no longer needs to executed.  For example, if a commit no longer",
+    "exists after a rebase, the task should be canceled and reported as an exception",
+    "preventing any future reruns."
+  ].join('\n')
+}, function(req, res) {
+  // Validate parameters
+  if (!checkParams(req, res)) {
+    return;
+  }
+
+  var ctx = this;
+
+  var taskId        = req.params.taskId;
+  var runId         = parseInt(req.params.runId);
+  var reason        = req.body.reason;
+
+  return ctx.Task.load(taskId).then(function(task) {
+    // if no task is found, we return 404
+    if (!task || !task.runs[runId]) {
+      return res.status(404).json({
+        message:  "Task not found or already resolved"
+      });
+    }
+
+    var workerGroup = task.runs[runId].workerGroup;
+    var workerId    = task.runs[runId].workerId;
+
+    // Authenticate request by providing parameters
+    if(!req.satisfies({
+      workerGroup:  workerGroup,
+      workerId:     workerId
+    })) {
+      return;
+    }
+
+    // Resolve task run
+    return ctx.Task.resolveTask(
```

Comment:
Unfortunately it seems `Task.resolveTask` only works if the run is `running`. Which might not be the case.